### DISCUSSION TITLE
musikcube 0.99.0

### DISCUSSION THREAD

**Discussion by clangen** [clangen/musikcube#560] [General]
## changes

* completely removed `boost` as a dependency across all projects; this is a huge milestone in the project, and will make it much easier to distribute stable builds moving forward.
* fixed issues with usernames on Windows that contain non-latin characters
* fixed issues displaying paths with non-latin characters in some parts of the app.
* fixed issue with directory scanning that may lead to incomplete indexing
* fixed issue where user's home directory may not be resolved correctly.
* reverted `ViewPager2` changes in `musikdroid`, as there is a bug that is preventing fragments from getting re-initialized properly during the resume cycle.
* updated Simplified Chinese translation (@UM-Li)
* added Czech translation (@khagaroth)
* upgraded to c++-17 runtime

<hr /><em>This discussion was created from the release <a href='https://github.com/clangen/musikcube/releases/tag/0.99.0'>musikcube 0.99.0</a>.</em>

**Comment by clangen** [comment]
While there are not many user-facing changes here, version `0.99.0` marks a significant milestone in `musikcube` development: the `boost` library has finally been removed from the project, and related functionality has been re-implemented by hand (or by using standard C++ language features). This is something I've been chipping away at slowly at for the past 5 years or so, and has finally made it across the finish line.

While the `boost` library provides a lot of useful functionality for developers, it was a major thorn in my side because it caused the app to stop functioning properly for users on UNIX operating systems relatively regularly.

**I would not be surprised if we see a number of new bugs come out of this release due to the magnitude of the change, but long-term it will make the project much more stable for end users, and easier to build for developers interested in contributing.**. 

If you see any weird behaviors or regressions, please make sure to let me know in this thread or open a ticket.

**Comment by clangen** [comment]
### Also note that the next non-patch release will likely be `3.0.0`.

Version `1.0` was originally released in 2007.
* https://musikcube.com/old/

Development of version `2.0` started sometime in 2009, I believe, but was never fully realized; I had a job that cited a conflict of interest and prevented me from contributing further.
* https://github.com/clangen/musikcube/releases/tag/mC2dm3

In about 2015, a while after changing employers, I resumed development. Unfortunately, the `2.0` code had been sitting dormant for more than half a decade and had started to rot, and most of the original contributors had moved on. At the time I wanted to build a new version just for myself, so I started hacking on it, eventually releasing a `0.2.1` version. I should have called it `3.2.1` or something, but didn't, because I never thought anyone would actually use it.
* https://github.com/clangen/musikcube/releases/tag/0.2.1

So, to avoid confusion with the original `1.0` release (and the never-finished `2.0` release), the next major version will be `3.0.0`.

**Comment by hrxn** [comment]
Wow, that's great news. Kudos on achieving this major milestone! 😄

**Comment by extrowerk** [comment]
Great. Just building the latest master on OpenBSD. Sadly the cmake tries to link against libtinfo, but commenting that section out the build proceeds. Will create a ticket...

**Comment by clangen** [comment]
Thanks for the heads up @extrowerk. I don't currently have an OpenBSD installation to test against, so I'll see if I can get a VM up and running here soon.

**Comment by clangen** [comment]
@extrowerk I just pushed a change to `master` that should hopefully fix this issue. If you have a chance to check it out, let me know if you're able to compile and run again.

I suspect there may be a similar issue with other BSD variants; I'll give FreeBSD a try sometime in the next couple days.

**Comment by extrowerk** [comment]
Thanks, now it builds just fine on OpenBSD!
My server is however in on FreeBSD and it is installed from ports, and the port is not yet updated, so now my client and server is incompatible. :(

**Comment by Brus67** [comment]
Hi @clangen
Distro: Linux Mint 20.3 Una base: Ubuntu 20.04 focal  
Kernel: 5.4.0-132-generic x86_64

Just updated and sadly I'm no longer able to play .wma files, errors listed below:

09:51:14 [info] [GaplessTransport] starting track at /mnt/sda2/home/user/Music/Aladdin Sane/01 Watch That Man.wma
09:51:14 [info] [Player] new instance created
09:51:14 [info] [GaplessTransport] stop
09:51:14 [info] [Stream] opening /mnt/sda2/home/user/Music/Aladdin Sane/01 Watch That Man.wma
09:51:14 [info] [LocalFileStream] opening file: /mnt/sda2/home/user/Music/Aladdin Sane/01 Watch That Man.wma
09:51:14 [info] [ffmpegdecoder] parsing data stream...
09:51:14 [info] [ffmpegdecoder] found audio stream!
09:51:14 [info] [ffmpegdecoder] resolved codec: Windows Media Audio 2
09:51:14 [info] [Streams] found a decoder for /mnt/sda2/home/user/Music/Aladdin Sane/01 Watch That Man.wma
09:51:14 [warning] [ffmpegdecoder] swr_convert_frame() failed: Output changed
09:51:14 [warning] [ffmpegdecoder] swr_convert_frame() failed: Output changed
09:51:14 [warning] [ffmpegdecoder] swr_convert_frame() failed: Output changed

Reverting for now, any ideas or suggestions would be appreciated,
Thanks.

**Comment by clangen** [comment]
@extrowerk you can disable that error by going to `settings` > `advanced settings` > `RemoteLibraryIgnoreVersionMismatch` -> `true`.

It should be safe in this case, but you do need to be careful keeping it enabled. :)

**Comment by clangen** [comment]
@Brus67 I'll look into it -- can you let me know what version you were running previously? I wouldn't have suspected `.wma` playback to suddenly break in this particular release, as it shouldn't have been touched. 

Edit: OK, the bug has been fixed here: https://github.com/clangen/musikcube/commit/696ffc8445beb9bbcd492e91373a404530c82124 and will be part of the next release.

**Reply by Brus67** [reply]
@clangen 
Thanks a lot, I really appreciate it.
